bin/fin

```diff
@@ -6287,6 +6344,631 @@ set_project_root_permissions ()
 	is_root && chown -R 1000:1000 "$PROJECT_ROOT"
 }
 
+# Pull main command for interacting with Hosting Providers.
+provider_pull ()
+{
+	eval $(parse_params "$@")
+	if [[ "${HOSTING_PLATFORM}" == "" || "${HOSTING_SITE}" == ""  ]]; then
+		echo-error "Hosting Provider not set up. Following variables must be set: HOSTING_PLATFORM and HOSTING_SITE"
+		exit 1
+	fi
+
+	HOSTING_PLATFORM=$(echo ${HOSTING_PLATFORM} | tr '[:upper:]' '[:lower:]')
+
+	# Check to see if a support platform is accepted.
+	if [[ "${HOSTING_PLATFORM}" != "acquia" ]] && [[ "${HOSTING_PLATFORM}" != "pantheon" ]] && [[ "${HOSTING_PLATFORM}" != "platformsh" ]] &&
+		[[ "${HOSTING_PLATFORM}" != "drush" ]] && [[ "${HOSTING_PLATFORM}" != "wp" ]]; then
```

Comment:
Can be simplified. See example https://github.com/docksal/docksal/blob/1d99f684d7dc4006453c09f74047703eb4dc4b6b/bin/fin#L6508

Reply:
Found a way around this.